afqinsight/nn/pt_models.py

```diff
@@ -0,0 +1,166 @@
+import math
+from dipy.utils.optpkg import optional_package
+from dipy.utils.tripwire import TripWire
+
+torch_msg = (
+    "To use afqinsight's pytorch models, you need to have pytorch "
+    "installed. You can do this by installing afqinsight with `pip install "
+    "afqinsight[tf]`, or by separately installing pytorch with `pip install "
+    "pytorch`."
+)
+
+torch, has_torch, _ = optional_package("torch", trip_msg=torch_msg)
+if has_torch:
+    import torch.nn as nn
+else:
+    # Since all model building functions start with Input, we make Input the
+    # tripwire instance for cases where pytorch is not installed.
+    Input = TripWire(torch_msg)
+    print("test")
+
+class mlp4_torch(nn.Module):
```

Comment:
Naming: no need to add `_torch` to class names here, because users will know where they are importing from. In fact, might be beneficial to use the exact same API here as in the tensorflow module. 

And that means that all the public interfaces here are functions (not classes!) that take inputs and give a model ready to fit as outputs. Or does that not make sense for Pytorch?

Reply:
One solution for this might be something like: 

```

class MLP4(...): 

   ... 

```

Followed by: 

```
def mlp4(...): 

   mymlp4 = MLP4(...)
   ... 

   return mymlp4
``